src/main/java/core/basesyntax/model/Action.java

```diff
@@ -0,0 +1,28 @@
+package core.basesyntax.model;
+
+public enum Action {
+    BALANCE("b"),
+    SUPPLY("s"),
+    PURCHASE("p"),
+    RETURN("r");
+
+    private String code;
```

Comment:
The field 'code' can be final since it is assigned only once in the constructor and should not be changed afterwards. Ensuring immutability where possible is a good practice.